physics/cires_ugwp.F90

```diff
@@ -172,6 +174,12 @@ subroutine cires_ugwp_run(do_ugwp, me,  master, im,  levs, ntrac, dtp, kdt, lonr
     real(kind=kind_phys),    intent(out), dimension(im)      :: tau_mtb, tau_ogw, tau_tofd, tau_ngw
     real(kind=kind_phys),    intent(out), dimension(im, levs):: gw_dudt, gw_dvdt, gw_dtdt, gw_kdis
     real(kind=kind_phys),    intent(out), dimension(im, levs):: dudt_mtb, dudt_ogw, dudt_tms
+
+    ! These arrays are only allocated if ldiag=.true.
+    real(kind=kind_phys),    intent(inout), dimension(im, levs) :: ldu3dt_ogw, ldv3dt_ogw, ldt3dt_ogw
```

Comment:
If these arrays are only allocated if ldiag3d (not ldiag) is true, then they should be assumed-size arrays (:,:) instead of having explicit dimensions.